src/test/perl/rbe_value_format.t

```diff
@@ -195,9 +194,17 @@ is($formatted_value, $FORMATTED_STRING_HASH, "String hash correctly formatted");
 $formatted_value = $rbe_fh->_formatAttributeValue(\%STRING_HASH,
                                                   LINE_FORMAT_KW_VAL,
                                                   LINE_VALUE_HASH,
+                                                  LINE_KEY_OPT_PREFIX_DASH,
+                                                 );
+is($formatted_value, $FORMATTED_STRING_HASH_2, "String hash correctly formatted (key prefixed with ')");
+$formatted_value = $rbe_fh->_formatAttributeValue(\%STRING_HASH,
+                                                  LINE_FORMAT_KW_VAL,
+                                                  LINE_VALUE_HASH,
                                                   LINE_VALUE_OPT_SINGLE,
                                                  );
-like($formatted_value, qr/$FORMATTED_STRING_HASH_SINGLE/, "String hash (1 key/val per line) correctly formatted");
+# When LINE_VALUE_OPT_SINGLE is set, formatAttributeValue() should return undef 
+# if the value cannot be interpolated as a string
+ok(!$formatted_value, "A hash passed as value cannot be interpolated as a string");
```

Comment:
the only way to test if formatAttributeValue returns undef (and not eg empty string or 0), is to use `ok(! defined(),`